### ISSUE TITLE
Invalid read in ewmh_init()

### ISSUE THREAD

**Issue by t-wissmann** [herbstluftwm/herbstluftwm#222]
Running winterbreeze with valgrind gives the following error/warning on startup:
```
==19722== Memcheck, a memory error detector
==19722== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==19722== Using Valgrind-3.14.0 and LibVEX; rerun with -h for copyright info
==19722== Command: ./herbstluftwm --verbose -c ../share/autostart
==19722== 
==19722== Invalid read of size 8
==19722==    at 0x4AEBA30: _XData32 (in /usr/lib/libX11.so.6.3.0)
==19722==    by 0x4AC5D99: XChangeProperty (in /usr/lib/libX11.so.6.3.0)
==19722==    by 0x150215: ewmh_init() (in /home/thorsten/git/winterbreeze/build/herbstluftwm)
==19722==    by 0x11F282: main (in /home/thorsten/git/winterbreeze/build/herbstluftwm)
==19722==  Address 0x5b5a708 is 0 bytes after a block of size 8 alloc'd
==19722==    at 0x4837DEF: operator new(unsigned long) (vg_replace_malloc.c:334)
==19722==    by 0x1501DD: ewmh_init() (in /home/thorsten/git/winterbreeze/build/herbstluftwm)
==19722==    by 0x11F282: main (in /home/thorsten/git/winterbreeze/build/herbstluftwm)
==19722== 

```

**Comment by t-wissmann** [comment]
With more debug flags:
```
==20541== Memcheck, a memory error detector
==20541== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==20541== Using Valgrind-3.14.0 and LibVEX; rerun with -h for copyright info
==20541== Command: ./herbstluftwm --verbose -c ../share/autostart
==20541== 
==20541== Invalid read of size 8
==20541==    at 0x4AEBA30: _XData32 (in /usr/lib/libX11.so.6.3.0)
==20541==    by 0x4AC5D99: XChangeProperty (in /usr/lib/libX11.so.6.3.0)
==20541==    by 0x1414DE: ewmh_init() (ewmh.cpp:114)
==20541==    by 0x12173B: main (main.cpp:925)
==20541==  Address 0x5b59038 is 0 bytes after a block of size 8 alloc'd
==20541==    at 0x4837DEF: operator new(unsigned long) (vg_replace_malloc.c:334)
==20541==    by 0x14148A: allocate (new_allocator.h:111)
==20541==    by 0x14148A: allocate (alloc_traits.h:436)
==20541==    by 0x14148A: _M_allocate (stl_vector.h:296)
==20541==    by 0x14148A: _M_range_initialize<int const*> (stl_vector.h:1466)
==20541==    by 0x14148A: vector (stl_vector.h:519)
==20541==    by 0x14148A: ewmh_init() (ewmh.cpp:113)
==20541==    by 0x12173B: main (main.cpp:925)
==20541== 
```

**Comment by t-wissmann** [comment]
Fixed by 056ff0eb1502e12b7bb6c73e7842241e2a8bd649
